app/controllers/cinema_halls_service/cinema_halls_service.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+class CinemaHallsService
+  def initialize(cinema_hall)
+    @cinema_hall = cinema_hall
+  end
+
+  def assign_seat_names
+    create_seats_array
+    @seats_a.each do |name|
+      @cinema_hall.seats.create(name: name)
+    end
+  end
+
+  def create_seats_array
+    rows_name = ('a'..'z').take(@cinema_hall.row_num).to_a
+    columns_name = (1..(@cinema_hall.column_num)).to_a
+    @seats_a = []
```

Comment:
Consider if you really need this array as the instance variable? Consider that this whole method can produce an array as the result and then you don't have to use instance variable but just return array from this method and use local variable to rely on it. 

The easiest way of doing it:

```ruby
  def create_seats_array
    rows_name = ('a'..'z').take(@cinema_hall.row_num).to_a
    columns_name = (1..(@cinema_hall.column_num)).to_a
    seats_array = []
    rows_name.each do |i|
      columns_name.each do |j|
        seats_array.append(i + j.to_s)
      end
    end
    
    seats_array
  end
```

Then you can just use result of this method in the public method:

```ruby
  def assign_seat_names
    seats_array = create_seats_array
    seats_array.each do |name|
      @cinema_hall.seats.create(name: name)
    end
  end
```